packages/web-frontend/src/sagas.js

```diff
@@ -633,23 +634,14 @@ function* fetchViewData(parameters, errorHandler) {
  *
  */
 function* fetchDashboardItemData(action) {
-  const { dashboardItemProject, infoViewKey } = action;
-
-  // If this dashboard item is a member of a module that requires extra work before fetching its
-  // data, allow the module to handle that work and return any extra url parameters
-  let prepareForDashboardItemDataFetch;
-  try {
-    // eslint-disable-next-line import/no-dynamic-require
-    const moduleSagas = require(`./${dashboardItemProject}/sagas`);
-    prepareForDashboardItemDataFetch = moduleSagas.prepareForDashboardItemDataFetch;
-  } catch (error) {
-    // the project is not associated with a module handling its own sagas, ignore
-  }
+  const { infoViewKey } = action;
+  const state = yield select();
+  const project = selectCurrentProjectCode(state);
 
   // Run preparation saga if it exists to collect module specific url parameters
   let extraUrlParameters = {};
-  if (prepareForDashboardItemDataFetch) {
-    extraUrlParameters = yield call(prepareForDashboardItemDataFetch);
+  if (project === 'disaster') {
+    extraUrlParameters = yield call(getDisasterDateRange);
```

Comment:
I felt that the existing pattern was a bit overengineered, especially given we haven't utilitised it for at least a year... So I took a conscious decision to simplify it

Reply:
Great call - hopefully this will avoid people getting confused by generically named variables!